### PR TITLE
feat/MSSDK-1504 checkout page edits

### PR DESCRIPTION
**Pull request by kkarasek**
### Description

It was pointed out that a successful coupon redemption does not trigger a change of an offer description. Additionally, several tweaks to enhance the overall customer experience were proposed. Specifically, these include communicating the regular price when a discount is no longer applicable or emphasizing the impact of a coupon.

### Updates

- fixed offer description following successful coupon code redemption
- fixed discounted price presentation for offers with a trial (_for instance, if an offer with a trial gets a 50% coupon discount, the OfficeCheckoutCard component should showcase the reduced price. **Important!** It does not apply to free offers, as an initial total price is presented in such cases_).
- added a vivid note about a coupon providing clear info regarding the applied discount
- added translations for newly introduced offer description variants and coupon note

### Screenshots

image

image

image

image

### Tbd
- add desc variants for seasonal subscriptions ✅
- and remaining billing periods (3m, 6m, annual) ✅
- try to simplify renderCouponNote() and generateDescriptionForCoupon()

### CODE REVIEW

File: src/components/CheckoutPriceBox/CheckoutPriceBox.js [hunk 1]
```diff
@@ -24,22 +27,107 @@ const CheckoutPriceBox = () => {
...
     country,
     totalPrice: finalPrice,
     currency
   } = useSelector(state => state.order.order);
 
+  const currencySymbol = currencyFormat[currency];
+
   const { t } = useTranslation();
 
+  const renderCouponNote = () => {
```
Comment (kkarasek) [inline src/components/CheckoutPriceBox/CheckoutPriceBox.js:45]: to be possibly simplified

File: src/components/OfferCheckoutCard/OfferCheckoutCard.tsx [hunk 2]
```diff
@@ -115,8 +119,83 @@ const OfferCheckoutCard = () => {
     );
   };
 
+  const generateDescriptionForCoupon = () => {
```
Comment (kkarasek) [inline src/components/OfferCheckoutCard/OfferCheckoutCard.tsx:122]: same here - possibly to be simplified

File: src/components/OfferCheckoutCard/OfferCheckoutCard.tsx [hunk 3]
```diff
@@ -115,8 +119,83 @@ const OfferCheckoutCard = () => {
...
+    );
+    let description;
+    let formattedDescription;
+    if (discountPeriods === 1) {
+      // non-standard period
+      if (period === '3months' || period === '6months') {
+        formattedDescription = `You will be charged ${currencySymbol}${formattedTotalPrice} (incl. ${taxCopy}) per ${periodMapper[
+          period as Period
+        ]
+          .chargedForEveryText as string} for the next billing period.<br/>After that time you will be charged a regular price of ${currencySymbol}${regularPrice}.`;
```
Comment (iwonakulacz) [inline src/components/OfferCheckoutCard/OfferCheckoutCard.tsx:137]: I think this this copy matches the logic of 1 discounted period. User is charged a discounted price now, and all next payments will be a regular price, so for me it should be more like:
You will be charged now ${currencySymbol}${formattedTotalPrice} (incl. ${taxCopy}) **now**. <br/>After **Period (formatted)** you will be charged a regular price of ${currencySymbol}${regularPrice}.`;

File: src/components/CheckoutPriceBox/CheckoutPriceBox.js [hunk 4]
```diff
@@ -11,11 +11,14 @@ import {
...
   StyledLabel,
-  StyledPriceWrapper
+  StyledPriceWrapper,
+  CouponNoteStyled
 } from './CheckoutPriceBoxStyled';
 
 const CheckoutPriceBox = () => {
-  const { customerPriceInclTax } = useSelector(state => state.offer.offer);
+  const { customerPriceInclTax, period } = useSelector(
+    state => state.offer.offer
```
Comment (m4tewoosh) [inline src/components/CheckoutPriceBox/CheckoutPriceBox.js:20]: We can use selectOnlyOffer from `offerSlice`
```suggestion
  selectOnlyOffer
```

File: src/components/OfferCheckoutCard/OfferCheckoutCard.tsx (hunk 2 above)
Comment (m4tewoosh) [inline src/components/OfferCheckoutCard/OfferCheckoutCard.tsx:122]:
```suggestion
  const getTrialDescription = () => {
```

File: src/components/OfferCheckoutCard/OfferCheckoutCard.tsx [hunk 5]
```diff
@@ -44,23 +44,27 @@ const OfferCheckoutCard = () => {
...
 
   const { t } = useTranslation();
 
-  const generateTrialDescription = () => {
-    const taxCopy = country === 'US' ? 'Tax' : 'VAT';
+  const taxCopy = country === 'US' ? 'Tax' : 'VAT';
 
+  const generateTrialDescription = () => {
     if (period === 'season' && freeDays) {
       const formattedDescription = `You will be charged {{currencySymbol}}{{grossPrice}} (incl. {{taxCopy}}) after {{freeDays}} days and will be renewed on the next season start date.`;
```
Comment (m4tewoosh) [inline src/components/OfferCheckoutCard/OfferCheckoutCard.tsx:69]:
```suggestion
      const description = `You will be charged {{currencySymbol}}{{grossPrice}} (incl. {{taxCopy}}) after {{freeDays}} days and will be renewed on the next season start date.`;
```

File: src/components/OfferCheckoutCard/OfferCheckoutCard.tsx (hunk 2 above)
Comment (m4tewoosh) [inline src/components/OfferCheckoutCard/OfferCheckoutCard.tsx:122]:
```suggestion
  const getCouponDescription = () => {
```

File: src/components/OfferCheckoutCard/OfferCheckoutCard.tsx [hunk 6]
```diff
@@ -115,8 +119,83 @@ const OfferCheckoutCard = () => {
...
   };
 
+  const generateDescriptionForCoupon = () => {
+    const formattedTotalPrice = formatNumber(totalPrice);
+    const regularPrice = calculateGrossPriceForFreeOffer(
+      offerPrice,
+      taxRate,
+      customerPriceInclTax
+    );
+    let description;
```
Comment (m4tewoosh) [inline src/components/OfferCheckoutCard/OfferCheckoutCard.tsx:129]:
```suggestion
  
```

File: src/components/OfferCheckoutCard/OfferCheckoutCard.tsx [hunk 7]
```diff
@@ -115,8 +119,83 @@ const OfferCheckoutCard = () => {
...
+    let formattedDescription;
```
Comment (m4tewoosh) [inline src/components/OfferCheckoutCard/OfferCheckoutCard.tsx:130]:
```suggestion
    let description;
```

File: src/components/CheckoutPriceBox/CheckoutPriceBox.js (hunk 4 above)
Comment (kkarasek) [reply src/components/CheckoutPriceBox/CheckoutPriceBox.js:20]: done

File: src/components/OfferCheckoutCard/OfferCheckoutCard.tsx (hunk 2 above)
Comment (m4tewoosh) [reply src/components/OfferCheckoutCard/OfferCheckoutCard.tsx:122]: Get rid of global `description` value, create `description` locally in every case and return whole translation. Like I proposed above renderCouponNote. We can also standarize those func names to `getXXXDescription` or `getXXXCopy`

Local description variable example:
```
if(xxx) {
 const description = `You will be charged ${currencySymbol}${formattedTotalPrice} (incl. ${taxCopy}) per ${period} for the next ${period}.<br/>After that time you will be charged a regular price of ${currencySymbol}${regularPrice}.`;
      return t(`subscription-desc-coupon-${period}`, description, {
        currencySymbol,
        formattedTotalPrice,
        taxCopy,
        period,
        regularPrice
      });
     }
```

File: src/components/CheckoutPriceBox/CheckoutPriceBox.js [hunk 8]
```diff
@@ -24,22 +27,107 @@ const CheckoutPriceBox = () => {
...
+          currencySymbol,
+          formattedDiscountAmount,
+          discountedPeriods,
+          period
+        }
+      );
+    }
+    return description;
+  };
+
```
[the remark this answers is not in the document]
Comment (kkarasek) [reply src/components/CheckoutPriceBox/CheckoutPriceBox.js:123]: i've introduced the changes - seems to be working fine

File: src/components/OfferCheckoutCard/OfferCheckoutCard.tsx (hunk 2 above)
Comment (kkarasek) [reply src/components/OfferCheckoutCard/OfferCheckoutCard.tsx:122]: done as well

File: src/components/OfferCheckoutCard/OfferCheckoutCard.tsx (hunk 3 above)
Comment (kkarasek) [reply src/components/OfferCheckoutCard/OfferCheckoutCard.tsx]: done

File: src/components/CheckoutPriceBox/CheckoutPriceBoxStyled.js [hunk 9]
```diff
@@ -73,3 +74,12 @@ export const StyledPriceWrapper = styled.li`
     width: 100%;
   }
 `;
+
+export const CouponNoteStyled = styled.div`
```
Comment (iwonakulacz) [inline src/components/CheckoutPriceBox/CheckoutPriceBoxStyled.js:78]: let's add a custom class name here, so publishers will be able to easily change the color of that text

Comment (kkarasek) [reply src/components/CheckoutPriceBox/CheckoutPriceBoxStyled.js]: just added

**Review by iwonakulacz — APPROVED** [review]

**Pull request merged**